test/built-ins/JSON/isRawJSON/invoked-as-a-fn.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2023 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-json.israwjson
+description: >
+  JSON.isRawJSON can be invoked as a function
+info: |
+  JSON.isRawJSON ( O )
+
+  1. If Type(O) is Object and O has an [[IsRawJSON]] internal slot, return true.
+  2. Return false.
+
+includes: [deepEqual.js]
+features: [json-parse-with-source]
+---*/
+
+function assertIsRawJson(rawJson, expectedRawJsonValue) {
+  assert.sameValue(null, Object.getPrototypeOf(rawJson));
+  assert(Object.hasOwn(rawJson, 'rawJSON'));
+  assert.deepEqual(['rawJSON'], Object.getOwnPropertyNames(rawJson));
+  assert.deepEqual([], Object.getOwnPropertySymbols(rawJson));
```

Comment:
We are trying to phase out `deepEqual` (#3476), but these work equally well with compareArray:
```suggestion
  assert.compareArray(Object.getOwnPropertyNames(rawJson), ['rawJSON']);
  assert.compareArray(Object.getOwnPropertySymbols(rawJson), []);
```
(note you will have to replace the includes with `[compareArray.js]`, also note that test262's assert functions place the actual value first then the expected value. V8's assert functions do it the other way around, so we have to switch them otherwise implementations will get confusing debug messages.)